code/model/modelanimation.cpp

```diff
@@ -10,7 +10,9 @@ namespace animation {
 
 	ModelAnimationState ModelAnimation::play(float frametime, ship* ship) {
 
-		switch (m_state) {
+		float& time = m_time[ship];
```

Comment:
Could you use a different name?  This could cause problems if we add/change a time library, or if we have some other `time` struct or definition.